src/main/java/apoc/algo/PageRank.java

```diff
@@ -55,6 +69,34 @@
                 Util.typesAndDirectionsToTypesArray((String) config.getOrDefault(SETTING_PAGE_RANK_TYPES, "")));
     }
 
+    @Procedure("apoc.algo.pageRankWithCypher")
+    @Description(
+            "")
+    public Stream<NodeScore> pageRankWithCypher(
+            @Name("nodes") List<Node> nodes,
+            @Name("config") Map<String, Object> config) {
+        Long iterations = (Long) config.getOrDefault(SETTING_PAGE_RANK_ITERATIONS, DEFAULT_PAGE_RANK_ITERATIONS);
+        String relCypher = (String)config.getOrDefault(SETTING_PAGE_RANK_CYPHER_REL, DEFAULT_PAGE_RANK_CYPHER_REL);
+        String nodeCypher = (String)config.getOrDefault(SETTING_PAGE_RANK_CYPHER_NODE, DEFAULT_PAGE_RANK_CYPHER_NODE);
+
+        boolean shouldWrite = (boolean)config.getOrDefault(SETTING_PAGE_RANK_WRITE, DEFAULT_PAGE_RANK_WRITE);
+
+        if (nodeCypher.equals("") || nodeCypher.isEmpty()) {
+            nodeCypher = DEFAULT_PAGE_RANK_CYPHER_NODE;
+        }
+
+        if (relCypher.equals("") || relCypher.isEmpty()) {
+            relCypher = DEFAULT_PAGE_RANK_CYPHER_REL;
+        }
+
+        PageRankArrayStorageParallelCypher pageRank = new PageRankArrayStorageParallelCypher(dbAPI, pool, nodeCypher, relCypher);
+        pageRank.compute(iterations.intValue());
+        if (shouldWrite) {
+            pageRank.writeResultsToDB();
+        }
+        return nodes.stream().map(node -> new NodeScore(node, pageRank.getResult(node.getId())));
```

Comment:
Ya ordering is probably too expensive. 
We could attempt to put them lazily into buckets. 

Optionally we could have an option that the node-query also returns degrees.

Von meinem iPhone gesendet

> Am 11.08.2016 um 07:23 schrieb Atul Jangra <notifications@github.com>:
> 
> In src/main/java/apoc/algo/PageRank.java:
> 
> > +        boolean shouldWrite = (boolean)config.getOrDefault(SETTING_PAGE_RANK_WRITE, DEFAULT_PAGE_RANK_WRITE);
> > +
> > +        if (nodeCypher.equals("") || nodeCypher.isEmpty()) {
> > +            nodeCypher = DEFAULT_PAGE_RANK_CYPHER_NODE;
> > +        }
> > +
> > +        if (relCypher.equals("") || relCypher.isEmpty()) {
> > +            relCypher = DEFAULT_PAGE_RANK_CYPHER_REL;
> > +        }
> > +
> > +        PageRankArrayStorageParallelCypher pageRank = new PageRankArrayStorageParallelCypher(dbAPI, pool, nodeCypher, relCypher);
> > +        pageRank.compute(iterations.intValue());
> > +        if (shouldWrite) {
> > +            pageRank.writeResultsToDB();
> > +        }
> > +        return nodes.stream().map(node -> new NodeScore(node, pageRank.getResult(node.getId())));
> In case when writeback is false and the user just want to see the result, shouldn't we be returning nodes?
> I've tried to follow the other pagerank implementation approach where we can project the nodes whose pagerank we want to see.
> 
> I can completely get rid of the projection and return information as you asked, runtimes for reading, computation and writing back.
> But what about the case when "write" is false?
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub, or mute the thread.
>